### PR TITLE
Di 22706 delta test

### PR DESCRIPTION
**Pull request by liorSela**

### CODE REVIEW

File: server-side/tests/api_tests/services/sync/services/sync-tests-service.ts [hunk 1]
```diff
@@ -22,7 +23,13 @@ export class SyncService {
     }
 
     async handleSyncData(syncRes:any,return_url: boolean = false){
-        return  return_url ? await this.getSyncDataFromUrl(syncRes.ResourcesURL) : await this.auditLogService.getSyncDataFromAudit(syncRes)
+        let data =await this.auditLogService.getSyncDataFromAudit(syncRes)
```
Comment (Chasky-H) [inline server-side/tests/api_tests/services/sync/services/sync-tests-service.ts:26]: why do we need the return_url parameter here?
Can't we just see in the data if it is returning a URL or Array?

File: server-side/tests/api_tests/services/sync/services/system-filter-service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    generateSystemFilterScheme(type: 'User'|'Account'|'None'){
+        const syncSchema:AddonDataScheme = {
+            Name: this.generateScehmaName(`_${type.toLowerCase()}`),
+            Type: "data",
+            SyncData: {
+                Sync: true
+            }
+        }
+        syncSchema.Fields= this.generateSchemaField(type)
```
Comment (Chasky-H) [inline server-side/tests/api_tests/services/sync/services/system-filter-service.ts:19]: missing space before =

File: server-side/tests/api_tests/services/sync/services/system-filter-service.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import GeneralService from "../../../../../potentialQA_SDK/server_side/general.service";
+import { SyncAdalService } from "./sync-adal-service";
+
+export class SystemFilterService extends SyncAdalService {    
+    private accountUUIDS: string[] =["3b5e29fb-ba1a-44ae-a84f-532028a9a28a","2d639aed-3a42-49b8-aaec-cdecc1fd2a37","f6521e90-7edc-49da-b9ed-28e7b252cfbc"]
+    private userUUIDS: string[] =["31cfbcba-08f0-4af8-900f-fcccde066af4","27f21174-29c7-4b70-bb72-b2d8f77b4bf6","cbfc3250-acab-46c6-a51b-67ed6442531d"]
+    private CORE_RESOURCES_ADDON_UUID ='fc5a5974-3b30-4430-8feb-7d5b9699bc9f'
+
+    generateSystemFilterScheme(type: 'User'|'Account'|'None'){
+        const syncSchema:AddonDataScheme = {
```
Comment (Chasky-H) [inline server-side/tests/api_tests/services/sync/services/system-filter-service.ts:12]: missing space after :

File: server-side/tests/api_tests/services/sync/services/system-filter-service.ts (hunk 2 above)
Comment (Chasky-H) [inline server-side/tests/api_tests/services/sync/services/system-filter-service.ts:19]: this row can be part of the object like:
```
Fields: this.generateSchemaField(type)
```

File: server-side/tests/api_tests/services/sync/services/system-filter-service.ts (hunk 3 above)
Comment (Chasky-H) [reply server-side/tests/api_tests/services/sync/services/system-filter-service.ts:12]: there are more like this..

File: server-side/tests/api_tests/services/sync/services/system-filter-service.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        const resourceField:AddonDataScheme['Fields'] = {
+            [`${type}_Field`]:{
+                Type: "Resource", 
+                Resource: resource, 
+                ApplySystemFilter: true,
+                AddonUUID: this.CORE_RESOURCES_ADDON_UUID
+            }
+        }
+        return type!='None' ? resourceField : nameField
```
Comment (Chasky-H) [inline server-side/tests/api_tests/services/sync/services/system-filter-service.ts:67]: try to be consistent with spacing

File: server-side/tests/api_tests/services/sync/test-commands/delta-test-command.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    async pushData(adalService: ADALTableService): Promise<any> {
+        // initializing adal schema with data, first property is number of fields
+        // second propety is number of characters in each field
+        const data = this.syncAdalService.generateFieldsData(1, 1)
+        
+        // upserting the same data for both schemes and saving again the time between second and first
+        await this.firstSchemeADALTable!.upsertRecord(data)
+        this.timeOfNewScheme = new Date()
```
Comment (simcha-waintrob) [inline server-side/tests/api_tests/services/sync/test-commands/delta-test-command.ts:35]: why you need to save the time twice?

File: server-side/tests/api_tests/services/sync/test-commands/delta-test-command.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        // tests
+        let oldSchemaName = this.firstSchemeADALTable!.schemaName
+        let newSchemaName = this.secondSchemeADALTable!.schemaName
+
+        expect(syncRes.syncResult).to.have.property('UpToDate').that.is.a('Boolean').and.is.equal(false)
+        expect(syncRes.syncResult).to.have.property('ExecutionURI').that.is.a('String').and.is.not.undefined
+
+        // getting schemes from sync response and validating that both second and first schemes is in the rsponse,
+        // because sync returns all of the schemas even if they are not for sync
+        let schemes = await this.syncDataResult.getSchemes()
```
Comment (simcha-waintrob) [inline server-side/tests/api_tests/services/sync/test-commands/delta-test-command.ts:69]: why both schemes are returned?

Comment (liorSela) [reply server-side/tests/api_tests/services/sync/test-commands/delta-test-command.ts:69]: empty schemes are returned, but the object are empty

**Review by simcha-waintrob — APPROVED** [review]

**Pull request merged**